allow setting payment means and payment means description, fixes #516

ThomasChr opened the pull request:
I tried to fix #516 and allow to set payment means code and payment means description.
I'm not a Java Guy, so be gentle when reviewing :-)

ThomasChr commented:
I also added reading of the payment means code and payment means information

jstaerk commented:
Is paymentMeansInformation/paymentMeansCode part of direct debit, of any payment means, or of the invoice? Would it npot be duplicated in the invoice when you read and in the direct debit when you write?

ThomasChr commented:
@jstaerk 
Thanks for taking a look!
Yeah, having that in DirectDebit and Invoice is a duplicate.
I killed it from Invoice, I think it's best at DirectDebit.

What do you think?

ThomasChr commented:
If I read the standard correctly payment means should be on invoice level. Should I change the code to be on Invoice level or is the direct debit approach good for now?

jstaerk commented:
good idea+implementation but why was the sepa debit code changed from 59 to 58?

ThomasChr commented:
> good idea+implementation but why was the sepa debit code changed from 59 to 58?

You absolutely right, sorry for that and thanks for taking a look!
Fixed the 58/59 and rebased the branch!

Paeprock commented:
Wouldn't this rather be needed in BankDetails instead of DriectDebit? Since DirectDebit should always have the direct debit payment means?
Because my invoices are not paid by direct debit but rather by a manual bank transfer.

ThomasChr commented:
@Paeprock 
I used the methods and hat the same idea. I tend to think the same.
@jstaerk Should I move the methods to BankDetails, what do you think?
